runtime/common/src/paras_registrar.rs

```diff
@@ -348,6 +356,45 @@ pub mod pallet {
 			NextFreeParaId::<T>::set(id + 1);
 			Ok(())
 		}
+
+		/// Add a manager lock from a para. This will prevent the manager of a
+		/// para to deregister or swap a para.
+		///
+		/// Can be called by Root, the parachain, or the parachain manager if the parachain is unlocked.
+		#[pallet::weight(T::DbWeight::get().reads_writes(1, 1))]
+		pub fn add_lock(origin: OriginFor<T>, para: ParaId) -> DispatchResult {
+			Self::ensure_root_para_or_owner(origin, para)?;
+			<Self as Registrar>::apply_lock(para);
+			Ok(())
+		}
```

Comment:
We somehow need to tell people to use this then.

Reply:
Do I understand correctly that a user has no guarantee that the chain will ever become trustless?  
Maybe we can add a block number to the config so that after that block the chain can be locked by anyone.

Reply:
I mean that is not our job nor the job of the relay chain to ensure that a chain is trustless. Maybe this is some very specific chain controlled by one entity. I think it would be fine if this entity keeps control all the time. I mean at some point there is maybe one parachain that controlls another parachain. If the first parachain is trustless, the second one is trustless as well :P

Reply:
Ah right, we cant make assumptions about the origin. So if its already trust-less, then its not an issue.

Reply:
My take on this is that we need to tell the UIs to put a "warning" flag on parachains which are unlocked, which inform users of the power that these parachains keep centralized. If the parachains want to remove this UI element, they will call this function.